homeassistant/components/sensor/openhardwaremonitor.py

```diff
@@ -0,0 +1,212 @@
+"""
+Support for Open Hardware Monitor Sensor Platform.
+"""
+
+from threading import Timer
+from datetime import timedelta
+import logging
+import json
+
+import urllib.request
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+STATE_MIN_VALUE = 'minimal_value'
+STATE_MAX_VALUE = 'maximum_value'
+STATE_VALUE = 'value'
+STATE_OBJECT = 'object'
+CONF_INTERVAL = 'interval'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=1)
+SCAN_INTERVAL = timedelta(seconds=5)
+
+OHM_VALUE = 'Value'
+OHM_MIN = 'Min'
+OHM_MAX = 'Max'
+OHM_CHILDREN = 'Children'
+OHM_NAME = 'Text'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=8085): cv.port
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Open Hardware Monitor platform."""
+    # update_json(config, add_devices)
+
+    data = OpenHardwareMonitorData(config)
+    add_devices(data.devices)
+
+
+class OpenHardwareMonitorDevice(Entity):
+    """Device used to display information from OpenHardwareMonitor."""
+
+    def __init__(self, data, name, path, obj, attributes):
```

Comment:
Problem is, this information is parsed on startup, but on an update this information isn't available anymore. I then use the supplied `path` argument to quickly parse the current/new values. I'll see what I can do to remove the parameters.